front/src/app/todo-list/todo-list.component.ts

```diff
@@ -48,21 +49,61 @@ export class TodoListComponent implements OnInit, OnDestroy {
 
   taskListInputValue: string = '';
 
+  resize$ = fromEvent(window, 'resize')
+
+  sidebarMode: MatDrawerMode = 'side'
+
+  sidebarOpen: string = 'true'
+
   private destroy$ = new Subject<void>();
 
   constructor(private taskListManager: TaskListManagerService, private apiService: ApiService, private changeDetectorRef: ChangeDetectorRef, private router: Router) {}
 
   ngOnInit(): void {
+    this.sidebarOpen = (localStorage.getItem('sidebarOpen') as string)
+    if (document.documentElement.clientWidth <= 750) {
+      this.sidebarMode = 'over'
+    }
+
+    this.resize$
+      .subscribe(() => {
+        if (document.documentElement.clientWidth >= 750) {
+          this.sidebarMode = 'side'
+          this.sidebarOpen = 'true'
+          this.changeDetectorRef.markForCheck();
+          return
+        }
+        
+        this.sidebarMode = 'over'
+        this.sidebarOpen = 'false'
+        this.changeDetectorRef.markForCheck();
+      })
+
     this.taskListManager.loadLists();
 
     this.taskListManager.listId$
       .pipe(takeUntil(this.destroy$))
       .subscribe(listId => {
+        if (!listId) {
+          localStorage.setItem('sidebarOpen', 'true')
+        }

```

Comment:
1. I'm wondering why you only change value in local storage but the current opened state?
2. better use `changeSidebarOpen` instead of duplicating code